plugins/src/webrtcsink/imp.rs

```diff
@@ -189,11 +194,30 @@ struct State {
     audio_serial: u32,
     video_serial: u32,
     streams: HashMap<String, InputStream>,
+    navigation_handler: Option<NavigationEventHandler>,
 }
 
+fn create_navigation_event(sink: gst_video::Navigation, msg: &str) {
+    let event: Result<gst_video::NavigationEvent, _> = serde_json::from_str(msg);
+
+    if let Ok(event) = event {
+        sink.send_event(event.build().structure().unwrap().to_owned());
```

Comment:
This seems a bit suboptimal API-wise. Why first build a `gst::Event` just to get a reference to the `gst::Structure` it contains and then creating a copy of that?

That probably should be improved a bit in the bindings. How is this usually done in C code?

Reply:
You can't do it in C, you need to use the GstNavigationInterface API as there is no API to create GstNavigation events in C. Using the navigation interface API here would make everything uglier as I already got the structures ready. I could add a `NavigationEvent.structure()` method in the bindings if you prefer.

Reply:
Added that new method here: https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/merge_requests/935